radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/AccountService.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.api.service;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.google.inject.Inject;
+import com.radixdlt.api.Rri;
+import com.radixdlt.api.data.BalanceEntry;
+import com.radixdlt.api.data.TxHistoryEntry;
+import com.radixdlt.api.store.ClientApiStore;
+import com.radixdlt.api.store.ClientApiStore.BalanceType;
+import com.radixdlt.api.store.TokenBalance;
+import com.radixdlt.application.Balances;
+import com.radixdlt.application.MyStakedBalance;
+import com.radixdlt.application.MyValidator;
+import com.radixdlt.application.ValidatorInfo;
+import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.engine.RadixEngine;
+import com.radixdlt.identifiers.AccountAddress;
+import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.identifiers.ValidatorAddress;
+import com.radixdlt.statecomputer.LedgerAndBFTProof;
+import com.radixdlt.utils.Pair;
+import com.radixdlt.utils.UInt256;
+import com.radixdlt.utils.UInt384;
+import com.radixdlt.utils.functional.Result;
+import com.radixdlt.utils.functional.Tuple.Tuple2;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.radixdlt.api.JsonRpcUtil.jsonArray;
+import static com.radixdlt.api.JsonRpcUtil.jsonObject;
+import static com.radixdlt.utils.functional.Tuple.tuple;
+
+public class AccountService {
+	private final RadixEngine<LedgerAndBFTProof> radixEngine;
+	private final ECPublicKey bftKey;
+	private final ClientApiStore clientApiStore;
+
+	@Inject
+	public AccountService(
+		RadixEngine<LedgerAndBFTProof> radixEngine,
+		@Self ECPublicKey bftKey,
+		ClientApiStore clientApiStore
```

Comment:
AccountService should not be dependent on ClientApiStore as we don't want all nodes to run another database

Reply:
Fixed